### PR TITLE
Make deleteGameNbrSaves a promise.

### PR DESCRIPTION
**Pull request by kberg**
Add tests. Add Localfileysstem implementation

### CODE REVIEW

File: src/database/LocalFilesystem.ts [hunk 1]
```diff
@@ -147,8 +147,20 @@ export class LocalFilesystem implements IDatabase {
     return this.getGame(gameId);
   }
 
-  deleteGameNbrSaves(_gameId: GameId, _rollbackCount: number): void {
-    console.error('deleting old saves not implemented.');
+  deleteGameNbrSaves(gameId: GameId, rollbackCount: number): Promise<void> {
+    if (rollbackCount <= 0) {
+      console.error(`invalid rollback count for ${gameId}: ${rollbackCount}`);
+      // Should this be an error?
+      return Promise.resolve();
```
Comment (bafolts) [inline src/database/LocalFilesystem.ts:154]:
```suggestion
      return;
```

Comment (bafolts) [reply src/database/LocalFilesystem.ts:154]: oh wait this isn't marked `async`. Could consider making this `async` function as well.

File: src/database/PostgreSQL.ts [hunk 2]
```diff
@@ -233,27 +233,20 @@ export class PostgreSQL implements IDatabase {
...
   }
 
-  deleteGameNbrSaves(game_id: GameId, rollbackCount: number): void {
+  async deleteGameNbrSaves(game_id: GameId, rollbackCount: number): Promise<void> {
     if (rollbackCount <= 0) {
-      console.error(`invalid rollback count for ${game_id}: $rollbackCount`);
-      return;
+      console.error(`invalid rollback count for ${game_id}: ${rollbackCount}`);
+      // Should this be an error?
+      return Promise.resolve();
```
Comment (bafolts) [inline src/database/PostgreSQL.ts:240]:
```suggestion
      return;
```

**Review by bafolts — APPROVED** [review]

File: src/database/LocalFilesystem.ts (hunk 1 above)
Comment (kberg) [reply src/database/LocalFilesystem.ts:154]: image

Can't. :D

**Pull request merged**